src/services/users.service.js

```
@@ -0,0 +1,55 @@
+let users = [];
+
+const getUsers = () => {
+  return users;
+};
+
+const createUser = (name) => {
+  const newUser = {
+    id: Date.now(),
+    name,
+  };
+
+  users.push(newUser);
+
+  return newUser;
+};
+
+const getUser = (id) => {
+  return users.find((user) => user.id === id);
+};
+
+const deleteUser = (id) => {
+  const userIndex = users.findIndex((user) => user.id === id);
+
+  if (userIndex === -1) {
+    return;
+  }
+
+  return users.splice(userIndex, 1);
+};
+
+const updateUser = ({ id, name }) => {
+  const user = users.find((person) => person.id === id);
```

`updateUser` finds the user with `users.find((person) => person.id === id)`. Same type-compatibility note applies: ensure `id` is the same type as stored IDs. Otherwise, updates may silently fail to find the target (requirement 3.6 expects 404 when not found).